dotCMS/src/main/java/com/dotcms/rendering/velocity/util/VelocityUtil.java

```diff
@@ -0,0 +1,607 @@
+	 * Changes $ and # to velocity escapes.  This is helps filter out velocity code injections.
+	 * @param s 
+	 * @return
+	 */
+	public static String cleanVelocity(String s) {
+		if (s==null) {
+			return null;
+		}
+		s=s.replace("$", "${esc.dollar}");
+		s=s.replace("#", "${esc.hash}");
+		return s;
+	}
+
+
+	
+	public String parseVelocity(String velocityCode, Context ctx){
+		VelocityEngine ve = VelocityUtil.getEngine();
+		StringWriter stringWriter = new StringWriter();
+		try {
+		   ve.evaluate(ctx, stringWriter, "VelocityUtil:parseVelocity", velocityCode);
+		}catch (Exception e) {
+		Logger.error(this,e.getMessage(),e);
+		}
+		return stringWriter.toString(); 
+		
+	}
+
+
+	public static String convertToVelocityVariable(final String variable, boolean firstLetterUppercase){
+		
+
+	      return (firstLetterUppercase) 
+	              ? StringUtils.camelCaseUpper(variable)
+	              : StringUtils.camelCaseLower(variable);
+		
+	}
+	
+	
+	public static Boolean isNotAllowedVelocityVariableName(String variable){
+		
+
+		String [] notallwdvelvars={"inode","type", "modDate", "owner", "ownerCanRead", "ownerCanWrite", "ownerCanPublish",
+				"modUser", "working", "live", "deleted", "locked","structureInode", "languageId", "permissions",
+				"identifier", "conHost", "conFolder", "Host", "folder"}; 
+		Boolean found=false;
+		for(String notallowed : notallwdvelvars){
+			 if(variable.equalsIgnoreCase(notallowed)){
+				 found=true;
+			 }
+			
+		}
+		return found;
+	}
+
+	/**
+	 * Returns a basic Velocity Context without any toolbox or request
+	 * response, session objects;
+	 * @return
+	 */
+
+	public static Context getBasicContext() {
+		Context context = new VelocityContext();
+		context.put("UtilMethods", new UtilMethods());
+		context.put("PortletURLUtil", new PortletURLUtil());
+		context.put("quote", "\"");
+		context.put("pounds", "##");
+		context.put("return", "\n");
+		context.put("velocityContext", context);
+		context.put("language", "1");
+		context.put("InodeUtils", new InodeUtils());
+		return context;
+	}
+	
+	
+	/**
+	 * Gets creates Velocity context will all the toolbox, user, host, language and request stuff
+	 * inside the map
+	 * @param request
+	 * @param response
+	 * @return
+	 */
+	public static ChainedContext getWebContext(HttpServletRequest request, HttpServletResponse response) {
+		return getWebContext(getBasicContext(), request, response);
+	}
+	
+	
+	public static ChainedContext getWebContext(Context ctx, HttpServletRequest request, HttpServletResponse response) {
+
+        if ( ctx == null ) {
+            ctx = getBasicContext();
+        }
+
+        // http://jira.dotmarketing.net/browse/DOTCMS-2917
+
+		//get the context from the request if possible
+        ChainedContext context;
+        if ( request.getAttribute( com.dotcms.rendering.velocity.Constants.VELOCITY_CONTEXT ) != null && request.getAttribute( com.dotcms.rendering.velocity.Constants.VELOCITY_CONTEXT ) instanceof ChainedContext ) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Split this 215 characters long line (which is greater than 150 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00103)